launch_testing/launch_testing/loader.py

```diff
@@ -231,14 +243,27 @@ def _partially_bind_matching_args(unbound_function, arg_candidates):
 
 
 def _give_attribute_to_tests(data, attr_name, test_suite):
+
+    def _warn_getter(self):
+        if not hasattr(self, '__warned'):
+            _logger.warning(
+                'Automatically adding attributes like self.{0} '
+                'to the test class will be deprecated in a future release.  '
+                'Instead, add {0} to the test method argument list to '
+                'access the test object you need'.format(attr_name)
+            )
+            setattr(self, '__warned', True)
+
+        return data
+
     # Test suites can contain other test suites which will eventually contain
     # the actual test classes to run.  This function will recursively drill down until
     # we find the actual tests and give the tests a reference to the process
 
     # The effect of this is that every test will have `self.attr_name` available to it so that
     # it can interact with ROS2 or the process exit coes, or IO or whatever data we want
     for test in _iterate_tests_in_test_suite(test_suite):
-        setattr(test, attr_name, data)
+        setattr(test.__class__, attr_name, property(fget=_warn_getter))
```

Comment:
@pbaughman careful with this, as the property applies to all tests that use that same class. Since attributes are always available, I guess it's fine in this case, though you may want to iterate over test classes instead.

Reply:
Yeah, basically. Do you think it's that risky?

Reply:
I think it's riskier than doing it this way, and also this whole method goes away once we stop injecting attributes into test classes, so the sub-optimal way of doing it is relatively short-lived.

I suppose I could do 
```
for test in _iterate_tests_in_test_suite(test_suite):
    if not hasattr(test.__class__, attr_name):
        setattr(test.__class__, attr_name, property(fget=_warn_getter))
```
That doesn't seem a whole lot better though, unless there's some huge performance penalty doing `setattr` or constructing a property object

Reply:
Why not using [`_iterate_test_classes_in_test_suite()`](https://github.com/ros2/launch/blob/f73bc56caa9c7b3c53b264475728392d0e1ce9cf/launch_testing/launch_testing/loader.py#L253-L258)?

Reply:
Sorry, yeah - because I forgot it was a thing.  Thank you for being a good reviewer